projects/packages/publicize/src/class-publicize.php

```diff
@@ -271,9 +279,64 @@ public function get_all_connections_for_user() {
 				}
 			}
 		}
+
+		if ( self::use_admin_ui_v1() && $include_connection_health && count( $connections_to_return ) > 0 ) {
+			$connections_to_return = $this->add_connection_test_results( $connections_to_return );
+		}
+
 		return $connections_to_return;
 	}
 
+	/**
+	 * To add the connection test results to the connections.
+	 *
+	 * @param array $connections The Jetpack Social connections.
+
+	 * @return array
+	 */
+	public function add_connection_test_results( $connections ) {
+		$path                   = sprintf( '/sites/%d/publicize/connection-test-results', absint( Jetpack_Options::get_option( 'id' ) ) );
+		$response               = Client::wpcom_json_api_request_as_user( $path, '2', array(), null, 'wpcom' );
+		$connection_results     = json_decode( wp_remote_retrieve_body( $response ), true );
+		$connection_results_map = array();
+		foreach ( $connection_results as $connection_result ) {
+			$connection_results_map[ $connection_result['connection_id'] ] = $connection_result['test_success'];
+		}
+
+		return array_map(
+			function ( $connection ) use ( $connection_results_map ) {
+				if ( isset( $connection_results_map[ $connection['connection_id'] ] ) ) {
+						$connection['connection_health'] = $connection_results_map[ $connection['connection_id'] ] ? 'Healthy' : '	';
+				}
+				return $connection;
+			},
+			$connections
+		);
+	}
+
+	/**
+	 * Get a connections for a user.
+	 *
+	 * @param int $connection_id The connection_id.
+
+	 * @return object
+	 */
+	public function get_connections_for_user( $connection_id ) {
```

Comment:
If this is only one connection should it be?
```suggestion
	public function get_connection_for_user( $connection_id ) {
```